engine/src/cubos/engine/tools/scene_editor/plugin.cpp

```diff
@@ -0,0 +1,247 @@
+#include <imgui.h>
+
+#include <cubos/core/ecs/query.hpp>
+#include <cubos/core/ecs/commands.hpp>
+
+#include <cubos/engine/tools/scene_editor/plugin.hpp>
+#include <cubos/engine/tools/entity_selector/plugin.hpp>
+#include <cubos/engine/imgui/plugin.hpp>
+
+#include <cubos/engine/assets/plugin.hpp>
+#include <cubos/engine/transform/plugin.hpp>
+#include <cubos/engine/scene/plugin.hpp>
+
+#include <cubos/engine/tools/asset_explorer/plugin.hpp>
+
+#include <cubos/core/settings.hpp>
+
+#include <cubos/engine/transform/position.hpp>
+
+using namespace cubos;
+using namespace cubos::engine::tools;
+using namespace cubos::engine;
+using cubos::core::ecs::Write;
+using cubos::core::ecs::Read;
+
+static void setSettings(Write<cubos::core::Settings> settings) {
+    settings->setBool("assets.io.enabled", true);
+    settings->setString("assets.io.path", "D:/Projetos/cubos/tools/tesseratos/assets");
+}
+
+static void closeScene(cubos::core::ecs::Commands& commands, SceneInfo& scene) {
+    for(const auto& [name, entity] : scene.entities) {
+        commands.destroy(entity);
+    }
+    scene.entities.clear();
+    for(auto& [name, subscene] : scene.scenes) {
+        closeScene(commands, subscene);
+    }
+    scene.scenes.clear();
+}
+
+static void placeEntity(std::string name, cubos::core::ecs::Entity handle, SceneInfo& scene){
+    auto split = name.find(".");
+    if(split == std::string::npos) {
+        scene.entities.emplace_back(name, handle);
+    }
+    else{
+        auto subsceneName = name.substr(0, split - 1);
+        for(auto& [sname, subscene] : scene.scenes) {
+            if(sname.compare(subsceneName) == 0) {
+                placeEntity(name.substr(split + 1), handle, subscene);
+                return;
+            }
+        }
+        SceneInfo subScene;
+        subScene.name = subsceneName;
+        auto& [_, subSceneHandle] = scene.scenes.emplace_back(subsceneName, subScene);
+        placeEntity(name.substr(split + 1), handle, subSceneHandle);
+    }
+}
+
+static void spawnScene(const Asset<Scene>& sceneToSpawn, cubos::core::ecs::Commands& commands, Read<Assets> assets, Write<SceneInfo> scene)
+{
+    closeScene(commands, *scene);
+    auto sceneRead = assets->read(sceneToSpawn);
+    auto builder = commands.spawn(sceneRead->blueprint);
+    for(const auto& [entity, name] : sceneRead->blueprint.getMap()) {
```

Comment:
You can use `builder.entities()` here and thus avoid having to add .getMap() and other map